### PR TITLE
[RFR] Improve Datagrid

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Distinguish non-sortable columns
- [x] Allow to style `<td>` (e.g. for number alignment)
- [x] Allow to style `<tr>` according to the value
- [x] Allow to set default sort order on list
- [x] [BC Break] The `filter` prop of the `<List>` component now expects an element rather than a component (`<List filter={<MyFilter/>} >` rather than `<List filter={MyFilter} >`) 
- [x] Add `perPage` and `pagination` props to the `<List>` component
- [x] Add `action` prop to customize list actions
- [x] [BC Break] The `title` prop of all view components now expect an element rather than a component  (`<List title={<MyTitle/>} >` rather than `<List title={MyTitle} >`) 
- [x] Add documentation

image

image

### CODE REVIEW

**Comment by fzaninotto** [comment]
Switching to RFR. Documentation will come in a future PR.

**Comment by fzaninotto** [comment]
And we're back to RFR. This time with documentation!

**Comment by fzaninotto** [comment]
Back to WIP: tests fail

File: src/mui/list/Actions.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import NavigationRefresh from 'material-ui/svg-icons/navigation/refresh';
+import CreateButton from '../button/CreateButton';
+
+const cardActionStyle = {
+    zIndex: 2,
+    display: 'inline-block',
+    float: 'right',
+};
+
+const Actions = ({ resource, Filter, displayedFilters, filterValues, hasCreate, basePath, showFilter, refresh }) => (
```
Comment (djhi) [inline src/mui/list/Actions.js]: I agree `<Filter ... />` is better than `<filter ... />` inside a component rendering code but it looks weird as a prop. Conventions are to name the prop in camelCase (`filter`) and you can use es6 to rename it inside the component with:

```js
const Actions = ({ resource, filter: Filter, displayedFilters, filterValues, hasCreate, basePath, showFilter, refresh }) => (
```

File: src/mui/list/List.js [hunk 2]
```diff
@@ -100,25 +149,15 @@ export class List extends Component {
     }
 
     render() {
-        const { filter, resource, hasCreate, title, data, ids, total, children, isLoading } = this.props;
+        const { Filter, Pagination = DefaultPagination, Actions = DefaultActions, resource, hasCreate, title, data, ids, total, children, isLoading } = this.props;
```
Comment (djhi) [inline src/mui/list/List.js]: Same remarks for `Filter`, `Pagination` and `Actions` here

**Pull request merged**